### PR TITLE
Implement expm1

### LINKED ISSUE #13: implement cosh
Source: https://git.musl-libc.org/cgit/musl/tree/src/math/cosh.c

### LINKED ISSUE #14: implement exp
Source: https://git.musl-libc.org/cgit/musl/tree/src/math/exp.c

### LINKED ISSUE #18: implement expm1
Source: https://git.musl-libc.org/cgit/musl/tree/src/math/expm1.c

### PR DESCRIPTION
**Pull request by Veykril**
~~Closes 13~~, closes #18 and ~~closes 14~~.

~~I wasn't sure where to put `__expo2(x: f64) -> f64` so I left it in `src/math/cosh.rs` for now.~~ Moved the function into it's own module.

Edit: Didn't see that `exp` was already done in a pull request, I'll take it out once #90 lands then.

### CODE REVIEW

**Comment by japaric** [comment]
LGTM but this will need to be rebased once #90 lands. After rebasing please leave a comment with the text "bors r=japaric" without the quotes to put this in [the bors queue](https://app.bors.tech/repositories/2772).

bors delegate+

**Comment by Veykril** [comment]
There is a small problem I have with `cosh` rn. Or at least something that confuses me. `k_expo2` will always return infinity( or NaN I think) , therefore `cosh` will always return infinity for certain inputs, but the travis test fails on `cosh` because for a value that it tests it expects a number that isnt infinity even though infinity is the result for me at least. I cant use the testsuite on my machine because I am running windows unfortunately

**Comment by Veykril** [comment]
@japaric should I just push the expm1 part for now and leave cosh aside?

**Comment by japaric** [comment]
@Veykril Sure, sgtm.

File: src/lib.rs [hunk 1]
```diff
@@ -432,15 +432,13 @@ pub trait F64Ext: private::Sealed {
...
 
-    #[cfg(todo)]
     fn exp_m1(self) -> Self;
 
     fn ln_1p(self) -> Self;
 
     #[cfg(todo)]
     fn sinh(self) -> Self;
 
-    #[cfg(todo)]
```
Comment (japaric) [inline src/lib.rs]: needs to be restored

File: src/lib.rs [hunk 2]
```diff
@@ -631,7 +628,6 @@ impl F64Ext for f64 {
         sinh(self)
     }
 
-    #[cfg(todo)]
```
Comment (japaric) [inline src/lib.rs]: needs to be restored

File: src/math/mod.rs [hunk 3]
```diff
@@ -9,8 +9,10 @@ macro_rules! force_eval {
 mod ceil;
 mod ceilf;
 mod cosf;
+mod cosh;
```
Comment (japaric) [inline src/math/mod.rs]: needs to be removed

File: src/math/mod.rs [hunk 4]
```diff
@@ -42,8 +44,10 @@ mod truncf;
 pub use self::ceil::ceil;
 pub use self::ceilf::ceilf;
 pub use self::cosf::cosf;
+pub use self::cosh::cosh;
```
Comment (japaric) [inline src/math/mod.rs]: needs to be removed

File: test-generator/src/main.rs [hunk 5]
```diff
@@ -702,10 +702,10 @@ f64_f64! {
     // cbrt,
     ceil,
     // cos,
-    // cosh,
+    cosh,
```
Comment (japaric) [inline test-generator/src/main.rs]: needs to be undone

**Comment by Veykril** [comment]
bors r=japaric

File: src/math/mod.rs [hunk 6]
```diff
@@ -72,8 +76,12 @@ pub use self::trunc::trunc;
 pub use self::truncf::truncf;
 
 mod k_cosf;
+mod k_expo2;
```
Comment (japaric) [inline src/math/mod.rs]: I think this needs to be removed too

File: src/math/mod.rs [hunk 7]
```diff
@@ -72,8 +76,12 @@ pub use self::trunc::trunc;
...
 mod k_sinf;
 mod rem_pio2_large;
 mod rem_pio2f;
 
-use self::{k_cosf::k_cosf, k_sinf::k_sinf, rem_pio2_large::rem_pio2_large, rem_pio2f::rem_pio2f};
+use self::{
```
Comment (japaric) [inline src/math/mod.rs]: I think this needs to be undone too

**Comment by Veykril** [comment]
my commit didnt pick everything up, my bad sorry

**Comment by japaric** [comment]
No worries.

bors r+

**Comment by Veykril** [comment]
bors r=japaric

**Pull request merged**